### ISSUE TITLE
Parent node importance?

### ISSUE THREAD

**Issue by ronikobrosly** [pgmpy/pgmpy#913]
Hello, 
Let's say you have a causal Bayesian Network DAG with CPDs already estimated. You set the evidence of a "target" node named X. Let's say X has 4 parent nodes. Is there a method for finding which of the 4 parent nodes exerts the most influence on X? In other words, a method for finding which of the parent nodes is most important.

Thanks!

**Comment by ankurankan** [comment]
Hi Roni,

I am not exactly sure what you are asking but what I understand is that you
are trying to find the correlation between the variables using the bayesian
network. And I really can't think of a way to relate/compute it. Maybe
these links might help in understanding: [1] [2].

[1]
https://stats.stackexchange.com/questions/189491/relationship-between-probability-distribution-and-correlation
[2]
https://stats.stackexchange.com/questions/189677/is-it-possible-to-generate-data-for-stochastic-process-with-specific-distributio

Cheers
Ankur

**Comment by ronikobrosly** [comment]
Thanks for the links and the super fast response Ankur! I was thinking something along these lines: https://www.rdocumentation.org/packages/bnlearn/versions/0.8/topics/arc.strength
or http://www.bnlearn.com/documentation/man/arc.strength.html

"Measure the strength of the probabilistic relationships expressed by the arcs (edges) of a Bayesian network..."

**Comment by ronikobrosly** [comment]
Actually, after digging a little deeper, it seems like this method is based on a bootstrapping approach which requires Bayesian Network structure learning: perturb the data and see how often an edge between two nodes is still there after perturbation (if it's hardly there after perturbation, it's a "weak" edge, but if it's still there many times after you perturb the data, it is "strong"). 

I think one could use this approach when you've already learned the structure and estimated the CPDs. You could randomly perturb one variable in the dataset many times, and see how it changes the relevant CPDs each time. If it changes wildly with each perturbation, maybe the edge you're looking at is weak.

**Comment by ankurankan** [comment]
I get it now. We don't already have anything like this implemented in pgmpy yet but I think it shouldn't be very hard to do.

**Comment by PARODBE** [comment]
Hi @ankurankan ,

one question, finally did you implemented a method to calculate the arc.strength as in bnlearn of R???

Thanks!

**Comment by ankurankan** [comment]
@PARODBE No, this hasn't been implemented yet.

**Comment by ankurankan** [comment]
Related #1684

**Comment by Manas-7854** [comment]
Hi @ankurankan, can we use the edge_strength function from the pr #1684 directly as a measure for parent node importance?

**Comment by ankurankan** [comment]
@Manas-7854 Yes, the edge strength now computes this.

**Comment by Manas-7854** [comment]
is there a need for a separate function(uses the edge strength function) that given a node gives out parent node importance?
shouldn't be hard to implement

**Comment by ankurankan** [comment]
@Manas-7854 I don't think so, the current method already does that. If you would like, you can add an example in the comment here as a reference.
